### PR TITLE
users: Add an list of recent logins to the account details

### PR DESCRIPTION
**Pull request by jelly**
Show a list of the 15 recent logins on the account details page, the last
login health card now includes a link to the account detail page.

Closes: #28

---

# Users: Login history

The user account details page now includes a list of your 15 most recent logins.

users-login-history

### CODE REVIEW

**Comment by jelly** [comment]
> Show a list of the 5 recent logins on the account details page, the last
> login health card now includes a link to the account detail page.
> 
> Closes: #28

That said, I'm not 100% sure it closes #28

File: pkg/users/account-logs-panel.jsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        }
+    }, [logins, name]);
+
+    return (
+        <Card id="account-logs">
+            <CardTitle>
+                <Text component="h2">{_("Account logs")}</Text>
+            </CardTitle>
+            <CardBody>
+                <DataList aria-label="Account log list" isCompact className="account-logs-list">
```
Comment (KKoukiou) [inline pkg/users/account-logs-panel.jsx:81]: aria-labels also need to be translated.

File: pkg/users/account-logs-panel.jsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+import React, { useState, useEffect } from 'react';
+
+import { Card, CardTitle, CardBody, DataList, DataListItem, DataListItemRow, DataListItemCells, DataListCell, Text } from '@patternfly/react-core';
```
Comment (KKoukiou) [inline pkg/users/account-logs-panel.jsx:23]: Do you have some mockup for this? 
The issue is that I am afriad you 're using the wrong component here.
If you want that header lines (table header) to be present you can't use data list, you need to use table, or our implementation of it, aka ListingTable.


When using this the mobile mode is broken, as headers appear as a cell. 
Screen Shot 2021-09-27 at 15 50 21

**Review by KKoukiou — CHANGES_REQUESTED** [review]

File: pkg/users/account-logs-panel.jsx (hunk 2 above)
Comment (jelly) [reply pkg/users/account-logs-panel.jsx:23]: > Do you have some mockup for this?

The mockup is here https://raw.githubusercontent.com/cockpit-project/cockpit-design/9a65b44ffa9cf60a6eaf02044e797d0046b77d6b/users/login-warning.png

> The issue is that I am afraid you 're using the wrong component here.
> If you want that header lines (table header) to be present you can't use data list, you need to use table, or our implementation of it, aka ListingTable.

Aha!

Comment (KKoukiou) [reply pkg/users/account-logs-panel.jsx:23]: It's a ListingTable with a contains-list class (we already use it) on the table body.

File: pkg/users/account-logs-panel.jsx (hunk 1 above)
Comment (jelly) [reply pkg/users/account-logs-panel.jsx]: done!

**Review by garrett — CHANGES_REQUESTED** [review]
Looks good overall!

I request these small tweaks:
- Heading should be "Login history"
- Table should be compact, show a few more entries if possible (possibly 10, 15)
- Alignment between page sections is not consistent, but it should be

Screenshot from 2021-09-28 16-12-05

**Comment by garrett** [comment]
Previous: 8690496
My changes: 3cd066a76

I squashed it into the original commit.

**Review by garrett — DISMISSED** [review]
With my little tweaks (@ 3cd066a76), I'm happy with this PR.

**Comment by garrett** [comment]
I've updated the screenshot and description as well.

File: pkg/systemd/overview-cards/lastLogin.jsx [hunk 3]
```diff
@@ -125,6 +129,15 @@ const LastLogin = () => {
                         {lastLoginText}
                     </FlexItem>
                     }
+                    {name &&
+                    <FlexItem>
+                        <Button variant="link" isInline
```
Comment (KKoukiou) [inline pkg/systemd/overview-cards/lastLogin.jsx:134]: Generally we should use `<a>` component when we get redirected to other pages, also inside cockpit. That said component='a' here with an href. 

However it's wrong all over cockpit, one more won't make a difference. We can fix it in a big batch in the future.

**Review by KKoukiou — APPROVED** [review]

**Review by KKoukiou — COMMENTED** [review]
there is one failure in https://logs.cockpit-project.org/logs/pull-16402-20210928-154922-172490ee-fedora-coreos/log.html#289

**Comment by garrett** [comment]
Aha! The spacing was fixed in wide modes, but it's collapsing oddly in narrower modes and I screenshot the narrower version. That's why I thought it was fixed (it is, in larger sizes) but the screenshot isn't.

I'll adjust the screenshot. We can fix things in follow-ups.

**Review by marusak — APPROVED** [review]

**Pull request merged**